docs/components/Autocomplete/Web.stories.tsx

```diff
@@ -1098,3 +1099,448 @@ function useCustomSectionOptions(
+              behavior: "smooth",
+              block: "nearest",
+              inline: "start",
+            });
+
+            break;
+          case KeyboardAction.Next: {
+            setHighlightedOptionIndex(newNextIndex);
+            const nextElement = menuRef?.children[newNextIndex];
+            const footerHeight = footerElement?.offsetHeight || 0;
+
+            if (nextElement) {
+              const rect = nextElement.getBoundingClientRect();
+              const menuRect = menuRef.getBoundingClientRect();
+
+              // If element is hidden behind footer
+              if (rect.bottom > menuRect.bottom - footerHeight) {
+                // Calculate exact scroll position needed
+                const scrollOffset =
+                  rect.bottom - (menuRect.bottom - footerHeight);
+                menuRef.scrollTop += scrollOffset;
+              } else {
+                nextElement.scrollIntoView({
+                  behavior: "smooth",
+                  block: "nearest",
+                  inline: "start",
+                });
+              }
+            }
+
+            if (newNextIndex === maxIndex) {
+              footerElement?.focus();
+            }
+            break;
+          }
+
+          case KeyboardAction.Select:
+            // Don't select the footer
+            highlightedOptionIndex < maxIndex &&
+              onOptionSelect(options[highlightedOptionIndex]);
+
+            menuRef?.children[highlightedOptionIndex]?.scrollIntoView?.({
+              behavior: "smooth",
+              block: "nearest",
+              inline: "start",
+            });
+
+            // If the footer is selected, click it
+            if (highlightedOptionIndex === maxIndex) {
+              footerElement?.click();
+            }
+            break;
+        }
+      },
+      [highlightedOptionIndex, options, onOptionSelect],
+    );
+    useCustomKeyboardNavigation({ onRequestHighlightChange });
+
+    const optionsToRender = options.map((option, index) => {
+      const label = option.status;
+      const status = option.status === "Active" ? "success" : "informative";
+
+      return (
+        <BaseMenuOption
+          addSeparators={true}
+          isHighlighted={index === highlightedOptionIndex}
+          onOptionSelect={onOptionSelect}
+          option={option}
+          key={option.value}
+        >
+          <Flex template={["grow", "shrink"]}>
+            <Flex align="start" template={["shrink", "grow"]}>
+              <Content spacing="minuscule">
+                {selectedOption === option && (
+                  <Icon name="checkmark" size="small" />
+                )}
+                <Text>{option.label}</Text>
+                <Text variation="subdued">{option.address}</Text>
+                <Text variation="subdued">{option.contact}</Text>
+              </Content>
+            </Flex>
+            <StatusLabel status={status} label={label} />
+          </Flex>
+        </BaseMenuOption>
```

Comment:
not a huge deal but something in the example is off with the layout causing it to exceed the width

<img width="737" alt="image" src="https://github.com/user-attachments/assets/88950df1-9be1-4abd-be54-67905233fd44" />